components/miscShared/fn_applyInsignia.sqf

```diff
@@ -5,61 +5,49 @@ params ["_unit"];
 
 LOCAL_ONLY(_unit);
 
-if !EXISTS(f_dict_insigniaColours) then
-{
-    _keyValues =
-    [
-        [str RED,       "insignia_GI_B_RED"],
-        [str ORANGE,    "insignia_GI_B_ORANGE"],
-        [str YELLOW,    "insignia_GI_B_YELLOW"],
-        [str GREEN,     "insignia_GI_B_GREEN"],
-        [str BLUE,      "insignia_GI_B_BLUE"],
-        [str CYAN,      "insignia_GI_B_CYAN"],
-        [str MAGENTA,   "insignia_GI_B_MAGENTA"],
-        [str PINK,      "insignia_GI_B_PINK"],
-        [str PURPLE,    "insignia_GI_B_PURPLE"],
-        [str WHITE,     "insignia_GI_B_WHITE"],
-        [str LIGHTGREY, "insignia_GI_B_LGREY"],
-        [str DARKGREY,  "insignia_GI_B_DGREY"],
-        [str BLACK,     "insignia_GI_B_BLACK"],
-        [str GREY,      "insignia_GI_B_GREY"],
-        [str BROWN,     "insignia_GI_B_BROWN"],
-        [str KHAKI,      "insignia_GI_B_TAN"]
-    ];
-
-    DICT_CREATE_VALS(f_dict_insigniaColours,_keyValues);
-
-};
-
 
 _faction = toLower (faction _unit);
 _unitType = _unit getVariable ["f_var_assignGear", ""];
+_insigniaVar = _unit getVariable ["f_var_insignia", ""];
 
 _insigniaClass = "";
 
+// Attempt to set insignia from f_var_insignia
+if (!(_insigniaVar isEqualTo "") && (_insigniaClass isEqualTo "" || _insigniaClass isEqualTo [])) then
```

Comment:
Extra complexity here in `(_insigniaClass isEqualTo "" || _insigniaClass isEqualTo [])` - I'm sure this is because of the way `DICT_GET` works.  I've now made a new `DICT_GET_DEFAULT` macro which makes this unnecessary - don't be afraid to make your own if it makes the code more readable.